src/components/GraphFilter/DurationDropdown.tsx

```diff
@@ -24,27 +25,43 @@ export class DurationDropdown extends React.Component<DurationDropdownProps, Dur
     [604800, '7 days'],
     [2592000, '30 days']
   ];
+  static InitialLabel = 'Select Duration';
+
   constructor(props: DurationDropdownProps) {
     super(props);
     this.state = {
-      currentDuration: props.initialDuration
+      currentDuration: props.initialDuration,
+      currentDurationName: this.getDurationName(props.initialDuration)
     };
   }
 
+  getDurationName = (duration: number) => {
+    let label = DurationDropdown.InitialLabel;
+    DurationDropdown.DurationIntervals.map(subarray => {
+      if (subarray[0] === Number(duration)) {
+        label = String(subarray[1]);
+      }
+    });
+    return label;
+  };
+
   onDurationChanged = (duration: number) => {
-    this.setState({ currentDuration: duration });
+    this.setState({ currentDuration: duration, currentDurationName: this.getDurationName(duration) });
     this.props.onClick(duration);
   };
 
   render() {
     return (
-      <DropdownButton id="duration" title="Duration" onSelect={this.onDurationChanged}>
-        {DurationDropdown.DurationIntervals.map(r => (
-          <MenuItem key={r[0]} active={r[0] === this.state.currentDuration} eventKey={r[0]}>
-            {r[1]}
-          </MenuItem>
-        ))}
-      </DropdownButton>
+      <div className="form-group">
+        <label style={{ paddingRight: '0.5em' }}>Duration:</label>
```

Comment:
If we are going to be doing this in multiple places, then we should create our own component for dropdowns that has an optional label property for it